app/models/manageiq/providers/kubernetes/container_manager/scanning/job.rb

```diff
@@ -215,6 +211,16 @@ def finish(*_args)
 
   alias_method :abort_job, :cleanup
 
+  def cancel(*args)
+    _log.info "Job Canceling"
+    if self.state != "canceling" # ensure change of states
+      signal :cancel
+    else
+      unqueue_all_signals
+      queue_signal(:abort_job, "Job Canceled", "cancel", *args)
```

Comment:
Currently abort_job is an alias for cleanup which in the end calls process_abort in the parent job.
We actually want to call process_cancel - that will make sure we behave like other jobs on cancel